include/librealsense2/dds/dds-device.h

```diff
@@ -51,11 +52,14 @@ class dds_device
 
     size_t num_of_sensors() const;
 
-    size_t foreach_sensor( std::function< void( const std::string& name ) > fn ) const;
+    size_t foreach_sensor( std::function< void( size_t sensor_index, const std::string & name ) > fn ) const;
 
     size_t foreach_video_profile( size_t sensor_index, std::function< void( const rs2_video_stream& profile, bool def_prof ) > fn ) const;
     size_t foreach_motion_profile( size_t sensor_index, std::function< void( const rs2_motion_stream& profile, bool def_prof ) > fn ) const;
 
+    void sensor_open( size_t sensor_index, const std::vector< rs2_video_stream > & profiles );
```

Comment:
Is this just an intermediate step, or the final API?
Maybe we should have a talk about where we're heading and sync up?

Reply:
This is a step towards control path messages and image streaming, but I'm not against refactoring after.
I'll set up a meeting with Nir and we'll discuss the design